From 5543649742b87590fb12916eb616c96fdee11be5 Mon Sep 17 00:00:00 2001
From: Rui Loura <rui@joyent.com>
Date: Fri, 9 Mar 2018 01:37:48 +0000
Subject: [PATCH] TRITON-606 ur-agent should be aware of alternate admin
 nictags Reviewed by: Cody Peter Mello <melloc@writev.io> Reviewed by: Trent
 Mick <trentm@gmail.com> Approved by: Cody Peter Mello <melloc@writev.io>

---
 ur-agent | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/ur-agent b/ur-agent
index be2c4c7..9a3dee0 100644
--- a/ur-agent
+++ b/ur-agent
@@ -247,7 +247,11 @@ function UrAgent(options) {
     function firstAdminMac() {
         var interfaces;
         var mac;
+        var admin_tag = 'admin';
 
+        if (self.sysinfo['Admin NIC Tag']) {
+            admin_tag = self.sysinfo['Admin NIC Tag'];
+        }
         interfaces = self.sysinfo['Network Interfaces'];
 
         for (var iface in interfaces) {
@@ -256,7 +260,7 @@ function UrAgent(options) {
             }
 
             var nic = interfaces[iface]['NIC Names'];
-            var isAdmin = nic.indexOf('admin') !== -1;
+            var isAdmin = nic.indexOf(admin_tag) !== -1;
             if (isAdmin) {
                 mac = interfaces[iface]['MAC Address'];
                 return mac;
-- 
2.21.0

